source/blog/2021-11-09-announcing-hanami-200alpha3.html.markdown

```diff
@@ -0,0 +1,163 @@
+excerpt: >
+  Streamlined source directories, more flexible settings, better integrated actions, and the beginning of a monthly release cadence.
+---
+
+Hello Hanami community! It’s Tim here again, and I’m delighted to announce the release of Hanami 2.0.0.alpha3!
+
+This release represents months of work on several foundational aspects of the framework, and brings:
+
+- Streamlined source directories for easier-to-navigate project file trees
+- Reworked application settings, leveraging a regular Ruby class to give you full flexibility with your setting definitions
+- Reworked application routes, to match the above
+- Improvements to view rendering within actions
+- Framework settings ported to dry-configurable
+
+## Streamlined source directories
+
+In our previous alpha release, we required you to structure your source files in a way that matched the typical Ruby conventions for loading from the `$LOAD_PATH`. So if you had a `main` slice, and a `Main::MyClass` component, it would need to be located at `slices/main/lib/main/my_class.rb`. This worked well enough, but it presented awkwardly deep directory trees with redundant names, especially given we expect all components defined within each slice to live within its own singular namespace anyway. This latter aspect was already made clear by the fact that we removed that redundant leading "main" from the component’s key, with it available from the slice as `Main::Slice["my_class"]`.
+
+So for this release, we made a major overhaul to our code loading to support our ideal source directory structure. Now in your `main` slice, your `Main::MyClass` component can be defined directly within it’s `lib/` directory, at `slices/main/lib/my_class.rb`, while still available from the slice as `Main::Slice["my_class"]`. This means one fewer directory to hop through, one less name to say when you’re communicating your source paths, and a much stronger signal that each slice should fully inhabit its own namespace.
+
+Along with this, we’ve created a new facility to support special categories of components to be kept in their own top-level directories. For this release, this is determined by `Hanami.application.config.component_dir_paths`, which defaults to `["actions", "repositories", "views"]`. This allows you to define an action like `Main::Actions::Posts::Index` in its own special directory, at `slices/main/lib/actions/posts/index.rb`, with the component accessible from the slice as `Main::Slice["actions.posts.index"]`.
+
+Together, these two changes should make finding and navigating both your slice’s business logic and its top-level entry points much easier.
+
+Given these changes rely heavily on our configured [Zeitwerk](https://github.com/fxn/zeitwerk) autoloader, we’ve decided to make the autoloader an always-on feature for Hanami 2. While we strive to make most things as configurable as possible behind the scenes, we now consider the autoloader a foundational pillar for the Hanami 2 experience.
```

Comment:
Pointing out the impossibility of turning off this feature, seems like bad news to me. We should just conclude this section with a reminder of a past good news.

```suggestion
Since the previous alpha, **[Zeitwerk](https://github.com/fxn/zeitwerk) autoloads your application code**.
```

Reply:
Ah, if you don't want that level of detail to be shared, then I might just delete the whole paragraph and not mention Zeitwerk at all. I'll save this detail for my own personal OSS status update :)